urwid/str_util.py

```diff
@@ -204,17 +236,25 @@ def calc_width(text: str | bytes, start_offs: int, end_offs: int) -> int:
     if start_offs > end_offs:
         raise ValueError((start_offs, end_offs))
 
-    utfs = isinstance(text, bytes) and _byte_encoding == "utf8"
-    unis = not isinstance(text, bytes)
-    if (unis and not SAFE_ASCII_RE.match(text)) or (utfs and not SAFE_ASCII_BYTES_RE.match(text)):
-        decode = [decode_one, decode_one_uni][unis]
+    if isinstance(text, str):
+        return sum(get_char_width(char) for char in text[start_offs:end_offs])
```

Comment:
here should be `wcwidth.wcswidth` usage, but then `calc_text_pos` may produce incompatible results.
Short details: 2 double-width calculated symbols may be rendered as single double width smile with accent (like `🤚🏻` is `🤚` + `accent`, both calculated as double width)

Reply:
Yes, 🤚🏻 ends with [EMOJI MODIFIER FITZPATRICK TYPE-1-2](https://codepoints.net/U+1f3fb), and there is also Zero-Width Joiner Emoji Sequences ([U+200D](https://codepoints.net/U+200d)), for example,
```
seq='\U0001f9d1\U0001f3ff\U0000200d\U0001f9b3'
>>> wcwidth.wcswidth(seq)  # right
2
>>> sum(map(wcwidth.wcwidth, seq))  # wrong
4
```